lib/web/mage/menu.js

```diff
@@ -20,8 +20,8 @@ define([
             responsive: false,
             expanded: false,
             showDelay: 42,
-            hideDelay: 300,
-            delay: 300,
+            hideDelay: 0,
```

Comment:
Unfortunately, we cannot simply remove the delay here. It's done for a purpose (I've described the main point in the original issue https://github.com/magento/magento2/issues/19652). We need another solution here.